@xen-orchestra/web-core/lib/components/ui/table-pagination/UiTablePagination.vue

```diff
@@ -0,0 +1,212 @@
+
+const { totalItems } = defineProps<{
+  totalItems: number
+  disabled?: boolean
+}>()
+
+const currentPage = ref(1)
+const itemsPerPage = ref(50)
+const pageSizeOptions = [10, 50, 100, 150, 200]
+
+const totalPages = computed(() => Math.ceil(totalItems / itemsPerPage.value))
+const startIndex = computed(() => (currentPage.value - 1) * itemsPerPage.value + 1)
+const endIndex = computed(() => Math.min(currentPage.value * itemsPerPage.value, totalItems))
+
+const isFirstPage = computed(() => currentPage.value === 1)
+const isLastPage = computed(() => currentPage.value === totalPages.value)
+
+const goToFirstPage = () => {
+  currentPage.value = 1
+}
+
+const goToPreviousPage = () => {
+  if (currentPage.value > 1) currentPage.value -= 1
+}
+
+const goToNextPage = () => {
+  if (currentPage.value < totalPages.value) currentPage.value += 1
+}
+
+const goToLastPage = () => {
+  currentPage.value = totalPages.value
+}
+
+const updatePagination = () => {
+  currentPage.value = 1
+}
+</script>
+
+<style scoped lang="postcss">
+.ui-table-pagination {
+  display: flex;
+  align-items: center;
+  gap: 0.8rem;
+
+  .buttons-content {
+    display: flex;
+    gap: 0.2rem;
+
+    .icon {
+      background-color: var(--color-neutral-background-primary);
+      border: 0.1rem solid var(--color-neutral-border);
+      font-size: 1rem;
+
+      &:hover {
+        border-color: var(--color-info-item-hover);
+        background-color: var(--color-info-background-hover);
+      }
+
+      &:active {
+        border-color: var(--color-info-item-active);
+        background-color: var(--color-info-item-active);
+      }
+
+      &:disabled {
+        background-color: var(--color-neutral-background-disabled);
+        border-color: transparent;
+
+        &:hover,
+        &:active {
+          background-color: var(--color-neutral-background-disabled);
+          border-color: transparent;
+        }
+      }
+    }
+  }
+
+  .label {
+    color: var(--color-neutral-txt-secondary);
+  }
+
+  .dropdown-wrapper {
+    position: relative;
+
+    .dropdown {
+      cursor: pointer;
+      padding: 2px 6px;
+      height: 2.6rem;
+      width: 4.8rem;
+      gap: 0.8rem;
+      appearance: none;
+      border-radius: 0.4rem;
+      color: var(--color-info-txt-base);
+      border: 0.1rem solid var(--color-neutral-border);
+      background-color: var(--color-neutral-background-primary);
+
+      &:hover {
+        border-color: var(--color-info-item-hover);
+        background-color: var(--color-info-background-hover);
+        color: var(--color-info-txt-hover);
+
+        + .icon {
+          color: var(--color-info-txt-hover);
+        }
+      }
+
+      &:disabled {
+        cursor: not-allowed;
+        background-color: var(--color-neutral-background-disabled);
+        color: var(--color-neutral-txt-secondary);
+        border-color: transparent;
+
+        + .icon {
+          color: var(--color-neutral-txt-secondary);
+        }
+      }
+
```

Comment:
In the design system, I see there is also the `:active` state, which seems missing here:
![image](https://github.com/user-attachments/assets/87ec2fd0-f2fd-4e3d-866e-cc827880a5c6)